src/Storages/MergeTree/MergeTreeDataPartChecksum.cpp

```diff
@@ -46,6 +46,10 @@ void MergeTreeDataPartChecksum::checkEqual(const MergeTreeDataPartChecksum & rhs
 
 void MergeTreeDataPartChecksum::checkSize(const IDataPartStorage & storage, const String & name) const
 {
+    /// Do not check inverted index files
```

Comment:
Could you please move l. 50+51 down to l. 55? (the exception for inverted indexes is the least important one of the three `if`'s)

Reply:
Kindly add a reason to the comment in l. 49, e.g. `/// Skip inverted index files, these have a default MergeTreeDataPartChecksum with file_size == 0`

Reply:
If you don't mind, I would like to understand how the database can end up with a checksum with file_size == 0 for inverted indexes (e.g. which methods are called in which order). Asking because I never encountered this error.

Reply:
What would be great if you could add a simple end-to-end stateless test (tests/queries/0_stateless/) which makes sure the situation doesn't occur again.